spring-cloud-gcp-core/src/main/resources/tracking-header.properties

```diff
@@ -0,0 +1 @@
+metrics.version=@parent.version@
```

Comment:
Can we scope this file to a package to avoid potential collisions with user code?
Also, maybe call the property `project.version`?

Reply:
Could you elaborate on what you mean? I'm not familiar with properties files. What do you mean by "scope this file to a package?"

Reply:
Well, our practice is do not have extra files on the matter, but just compile in the version into some class. The best sample is: https://github.com/spring-projects/spring-integration/blob/master/spring-integration-core/src/main/java/org/springframework/integration/config/xml/AbstractIntegrationNamespaceHandler.java#L47

But we can try some other solution, e.g. generate such a class during maven build. 

Saying that -1 from me for this extra file which may be lost during repackaging as you mention in the issue.

Reply:
I just mean putting it in a subdirectory like `resources/org/springframework/cloud/core/` like we did for the logging config files.

Reply:
@ChengyuanZhao On a second thought, I think @artembilan might be right. Depending on how people are repacking our code, this file can be lost, or generated differently. The safest thing might be to just hard-code the version in a Java class.

Reply:
There is this for your consideration: http://www.mojohaus.org/templating-maven-plugin/index.html